sample-code/src/orchestration.ts

```diff
@@ -1,133 +1,170 @@
+    // Define the language model to be used
     llm: {
-      model_name: 'gpt-4-32k',
+      model_name: 'gpt-4o',
       model_params: {}
     },
+    // Define the prompt
     templating: {
       template: [{ role: 'user', content: 'What is the capital of France?' }]
     }
   });
 
-  // Call the orchestration service.
-  const response = await orchestrationClient.chatCompletion();
-  // Access the response content.
-  return response.getContent();
+  // execute the request
+  const result = await orchestrationClient.chatCompletion();
+
+  // use getContent() to access the LLM response
+  logger.info(result.getContent());
+
+  return result;
 }
 
+const llmConfig: LlmModuleConfig = {
+  model_name: 'gpt-4o',
+  model_params: {}
+};
+
 /**
  * Ask about the capital of any country using a template.
- * @returns The message content from the orchestration service in the generative AI hub.
+ * @returns The orchestration service response.
  */
-async function orchestrationCompletionTemplate(): Promise<string | undefined> {
+export async function orchestrationTemplating(): Promise<OrchestrationResponse> {
   const orchestrationClient = new OrchestrationClient({
-    llm: {
-      model_name: 'gpt-4-32k',
-      model_params: {}
-    },
+    llm: llmConfig,
     templating: {
       template: [
+        // define "country" as variable by wrapping it with "{{? ... }}"
         { role: 'user', content: 'What is the capital of {{?country}}?' }
       ]
     }
   });
 
-  // Call the orchestration service.
-  const response = await orchestrationClient.chatCompletion({
+  return orchestrationClient.chatCompletion({
+    // give the actual value for the variable "country"
     inputParams: { country: 'France' }
   });
-  // Access the response content.
-  return response.getContent();
 }
 
+const template = { template: [{ role: 'user', content: '{{?input}}' }] };
+
 /**
- * Allow any user input and apply input and output filters.
- * Handles the case where the input or output are filtered:
- * - In case the input was filtered the response has a non 200 status code.
- * - In case the output was filtered `response.getContent()` throws an error.
- * @returns The message content from the orchestration service in the generative AI hub.
+ * Apply a content filter to LLM requests, filtering any hateful input.
  */
-async function orchestrationCompletionFiltering(): Promise<string | undefined> {
+export async function orchestrationInputFiltering(): Promise<void> {
+  // create a filter with minimal thresholds for hate and violence
+  // lower numbers mean more strict filtering
   const filter = buildAzureContentFilter({ Hate: 0, Violence: 0 });
   const orchestrationClient = new OrchestrationClient({
-    llm: {
-      model_name: 'gpt-4-32k',
-      model_params: {}
-    },
-    templating: {
-      template: [{ role: 'user', content: '{{?input}}' }]
-    },
+    llm: llmConfig,
+    templating: template,
```

Comment:
[pp] Consider naming the variables `llmConfig` => `llm` and `template` => `templating`, then you could do this:
```suggestion
    llm,
    templating,
```